generators/heroku/generator.mjs

```diff
@@ -709,7 +709,48 @@ export default class HerokuGenerator extends BaseGenerator {
 
             this.logger.log(chalk.bold('\nConfiguring Heroku'));
             await execCmd(`heroku config:set ${configVars}--app ${this.herokuAppName}`);
-            await execCmd(`heroku buildpacks:add ${buildpack} --app ${this.herokuAppName}`);
+            const { stdout: data } = await execCmd(`heroku buildpacks:add ${buildpack} --app ${this.herokuAppName}`);
+            if (data) {
+              this.logger.info(data);
+              // remote:  !     The following add-ons were automatically provisioned: . These add-ons may incur additional cost,
+              // which is prorated to the second. Run `heroku addons` for more info.
+              if (data.includes('Run `heroku addons` for more info.')) {
+                this.spawnCommand('heroku addons');
+              }
+
+              this.log('');
+              const prompts = [
+                {
+                  type: 'list',
+                  name: 'userDeployDecision',
+                  message: 'Continue to deploy?',
+                  choices: [
+                    {
+                      value: 'Yes',
+                      name: 'Yes, I confirmed',
+                    },
+                    {
+                      value: 'No',
+                      name: 'No, break (Recommended)',
+                    },
+                  ],
+                  default: 0,
+                },
+              ];
+
+              this.log('');
+              const props = await this.prompt(prompts);
+              if (props.userDeployDecision === 'Yes') {
+                this.log.info(chalk.bold('Continued deploying...'));
+              } else {
+                this.log(this.logger);
+                this.log.info(chalk.bold('You aborted deployment!'));
+                this.abort = true;
```

Comment:
```suggestion
                throw new Error('relevant message');
```

Reply:
Thanks for your suggestions, but here `throw new Error` will not abort program, `return` is needed.

<img width="794" alt="Screenshot 2023-04-23 at 19 33 38" src="https://user-images.githubusercontent.com/31362988/233837385-3cdd5b95-32ce-454f-a9ca-98662cdc5d05.png">

Reply:
Ok, it's inside a callback.